realm/realm-library/src/androidTest/java/io/realm/RealmSchemaTests.java

```diff
@@ -115,6 +115,87 @@ public void create_duplicatedNameThrows() {
     }
 
     @Test
+    public void createWithPrimaryKeyField_string() {
+        // Not nullable
+        realmSchema.createWithPrimaryKeyField("FooNonNull", "pkField", String.class, FieldAttribute.REQUIRED);
+        RealmObjectSchema objectSchema = realmSchema.getSchemaForClass("FooNonNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.STRING, objectSchema.getFieldType("pkField"));
+        assertFalse(objectSchema.isNullable("pkField"));
+
+        // Nullable
+        realmSchema.createWithPrimaryKeyField("FooNull", "pkField", String.class);
+        objectSchema = realmSchema.getSchemaForClass("FooNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.STRING, objectSchema.getFieldType("pkField"));
+        assertTrue(objectSchema.isNullable("pkField"));
+    }
+
+    @Test
+    public void createWithPrimaryKeyField_integer() {
+        // Not nullable
+        realmSchema.createWithPrimaryKeyField("FooNonNull", "pkField", int.class);
+        RealmObjectSchema objectSchema = realmSchema.getSchemaForClass("FooNonNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.INTEGER, objectSchema.getFieldType("pkField"));
+        assertFalse(objectSchema.isNullable("pkField"));
+
+        // Nullable
+        realmSchema.createWithPrimaryKeyField("FooNull", "pkField", Integer.class);
+        objectSchema = realmSchema.getSchemaForClass("FooNull");
+        assertEquals("pkField", objectSchema.getPrimaryKey());
+        assertEquals(RealmFieldType.INTEGER, objectSchema.getFieldType("pkField"));
+        assertTrue(objectSchema.isNullable("pkField"));
+    }
+
+    @Test
+    public void createWithPrimaryKeyField_invalidClassNameThrows() {
+        String[] names = { null, "", TestHelper.getRandomString(57) };
+
+        for (String name : names) {
+            try {
+                realmSchema.createWithPrimaryKeyField(name, "pkField", int.class);
+                fail();
+            } catch (IllegalArgumentException ignored) {
+            }
+            assertFalse(String.format("'%s' failed", name), realmSchema.contains(name));
+        }
+    }
+
+    @Test
+    public void createWithPrimaryKeyField_invalidFieldNameThrows() {
+        String[] fieldNames = new String[] { null, "", "foo.bar", TestHelper.getRandomString(65) };
```

Comment:
We must make sure that the return value of `TestHelper.getRandomString(65)` does not contain `.`.